xep-0388.xml

```diff
@@ -195,71 +300,187 @@
 </continue>
 ]]></example>
       <p>After the final octet of the first &lt;continue> element, any SASL security layer negotiated in the preceding exchange SHALL be immediately in effect.</p>
-      <p>Clients respond with a &lt;next/> element, which has a single mandatory attribute of "task", containing the selected task name, and contains an OPTIONAL base64 encoded initial response contained in an &lt;initial-response> element.</p>
+      <p>Clients respond with a &lt;next/> element, which has a single mandatory attribute of "task", containing the selected task name, and MAY contain any other element defined in a future protocol defining this task.</p>
+      <p>The concrete elements exchanged for each task after the &lt;next/> have to be defined in a future protocol defining a concrete task. Each task MUST end either by the server sending a &lt;failure/> element, if the task failed, a &lt;continue/> element, if the task was completed successfully and the server requests the client to perform a new task, or a &lt;success/> element, indicating that the task was completed successfully and no further tasks are needed.</p>
```

Comment:
I designed tasks to be syntactically aligned with SASL mechanisms, partly in the hope that other protocols might adopt them, and partly because it allowed for an easily pluggable system on the endpoints (like SASL mechanisms themselves), and partly because it meant that there are only SASSL2 top level elements in flow during the entirety of the SASL2 exchange, making a few bits of housekeeping simpler.

The fact you've elected to switch to XML is in itself fine, but it does mean that the exchange now introduces arbitrary elements, and the SASL2 code itself needs to be more aware of what tasks may be in flight.

I'm not averse to using XML here for tasks; I don't honestly have much hope for (for example) LDAP or IMAP using them. But I do think that means you'll need other framing to handle the now-discarded properties.